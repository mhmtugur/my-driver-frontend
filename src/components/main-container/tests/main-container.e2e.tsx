import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';

describe('main-container', () => {
  let page: E2EPage;
  let element: E2EElement;
  let searchInput;
  beforeEach(async () => {
    page = await newE2EPage({ html: '<main-container></main-container>' });
    element = await page.find('main-container');
    searchInput = await page.find('#searchInput');
    await setupSelectLocation(page, searchInput);
  });

  it.skip('should render offers', async () => {
    const formElement = await page.find('#search-form');
    formElement.triggerEvent('onSubmit');
    await page.waitForChanges();
    expect(element.innerHTML).toMatchSnapshot();
  })
});

const setupSelectLocation = async (page, searchInput) => {
  searchInput.press('b');
  searchInput.press('e');
  searchInput.press('r');
  await page.waitForChanges();
  let listItem = await page.find('.list-item');
  await listItem.click();
  await page.waitForChanges();
}