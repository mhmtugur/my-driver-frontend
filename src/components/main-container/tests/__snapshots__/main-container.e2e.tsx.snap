// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main-container should render offers 1`] = `"<div><top-banner class=\\"hydrated\\"><nav class=\\"nav-container\\"><div class=\\"nav-wrapper container\\"><img class=\\"brand-logo\\" src=\\"../../assets/image/sixt-logo.svg\\"></div></nav></top-banner><div class=\\"container\\"><div class=\\"row\\"><search-bar class=\\"hydrated\\"><form id=\\"search-form\\"><div class=\\"row\\"><div class=\\"col m5 s12\\"><location-container class=\\"hydrated\\"><div><div><location-search class=\\"hydrated\\"><div class=\\"input-field\\"><i class=\\"material-icons prefix\\">location_on</i><input id=\\"searchInput\\" class=\\"validate truncate\\" required=\\"\\" type=\\"text\\" autocomplete=\\"off\\" placeholder=\\"City,airport,location... etc.\\"></div></location-search></div></div></location-container></div><div class=\\"col m4 s12\\"><date-picker class=\\"hydrated\\"><div class=\\"input-field\\"><i class=\\"material-icons prefix\\">event_on</i><input type=\\"datetime-local\\" required=\\"\\" min=\\"2019-05-28T06:54\\" max=\\"2020-05-28T06:54\\"></div></date-picker></div><div class=\\"col m2 s12\\"><duration-dropdown class=\\"hydrated\\"><div class=\\"input-field\\"><select class=\\"browser-default\\" required=\\"\\"><option value=\\"\\" disabled=\\"\\">Duration (Hours)</option><option value=\\"1\\">1 Hours</option><option value=\\"2\\">2 Hours</option><option value=\\"3\\">3 Hours</option><option value=\\"4\\">4 Hours</option><option value=\\"5\\">5 Hours</option><option value=\\"6\\">6 Hours</option><option value=\\"7\\">7 Hours</option><option value=\\"8\\">8 Hours</option><option value=\\"9\\">9 Hours</option><option value=\\"10\\">10 Hours</option><option value=\\"11\\">11 Hours</option><option value=\\"12\\">12 Hours</option><option value=\\"13\\">13 Hours</option><option value=\\"14\\">14 Hours</option><option value=\\"15\\">15 Hours</option><option value=\\"16\\">16 Hours</option><option value=\\"17\\">17 Hours</option><option value=\\"18\\">18 Hours</option><option value=\\"19\\">19 Hours</option><option value=\\"20\\">20 Hours</option><option value=\\"21\\">21 Hours</option><option value=\\"22\\">22 Hours</option><option value=\\"23\\">23 Hours</option><option value=\\"24\\">24 Hours</option></select></div></duration-dropdown></div><div class=\\"col m1 s12\\"><search-button class=\\"hydrated\\"><div class=\\"input-field\\"><button class=\\"btn waves-effect waves-light\\">Search</button></div></search-button></div></div></form></search-bar></div><div class=\\"row\\"><offer-card-list class=\\"hydrated\\"><div></div></offer-card-list></div></div></div>"`;
